File: retrocast/internal/models/attachment.go
package models

type Attachment struct {
	ID          int64  `json:"id,string"`
	MessageID   int64  `json:"message_id,string"`
	Filename    string `json:"filename"`
	ContentType string `json:"content_type"`
	Size        int64  `json:"size"`
	StorageKey  string `json:"storage_key"`
}

File: retrocast-client/Retrocast/Sources/Models/Attachment.swift
import Foundation

struct Attachment: Codable, Identifiable, Hashable, Sendable {
    let id: Snowflake
    let messageID: Snowflake
    let filename: String
    let contentType: String
    let size: Int64
    let url: String

    enum CodingKeys: String, CodingKey {
        case id
        case messageID = "message_id"
        case filename
        case contentType = "content_type"
        case size
        case url
    }
}
